db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -133,9 +139,25 @@ class AppDataManager @Inject constructor(
                             val result = progressDataState.result as ArticResult<*>
                             if (result.data.isNotEmpty() && result.data[0] is ArticExhibition) {
                                 appDatabase.exhibitionDao.clear()
-                                @Suppress("UNCHECKED_CAST")
-                                appDatabase.exhibitionDao.updateExhibitions((result as ArticResult<ArticExhibition>).data)
-
+                                /**
+                                 * Update the sort order of the exhibitions according to the ArticExhibitionCMS
+                                 */
+                                appDatabase.exhibitionCMSDao
+                                        .getAllCMSExhibitions()
+                                        .subscribe {
+                                            @Suppress("UNCHECKED_CAST")
+                                            val list = (result as ArticResult<ArticExhibition>).data
```

Comment:
this cast is strange. why not cast it further up to the top where its used?

Reply:
We can cast it there ... but at this time we don't really need to use it anywhere higher, so we cast it when necessary